### PR TITLE
merge upstream sandialabs

### PR DESCRIPTION
**Pull request by jacobmerson**
This pull request includes changes necessary to merge the upstream sandialabs into the SCOREC fork.

As part of this, there were some major changes to the storage of the `rc_tags`.

### CODE REVIEW

**Comment by jacobmerson** [comment]
@joshia5 apologies that this is a bit of a big pull request since it has all of the merged changes from https://github.com/sandialabs/omega_h.

Please take a look at what's here especially the changes to the rc stuff that you have a good handle on.

**Comment by jacobmerson** [comment]
Interesting...tests show a segfault with gcc (although I see no issue on the SCOREC machine with gcc@11.2.0 with address sanitizer. I will try building with gcc@7.5.0

**Comment by jacobmerson** [comment]
Looks like the issue is related to setting the CXX_STANDARD to 17. Somehow this causes a failure with gcc/7.5.0

**Comment by jacobmerson** [comment]
@joshia5 All set.

**Comment by joshia5** [comment]
at a first glance isn't there supposed to be like a 100 commits from upstream or is there something broken in the commit history?

File: src/Omega_h_mesh.hpp [hunk 1]
```diff
@@ -127,92 +127,113 @@ class Mesh {
...
- * memory or not.
- */
-  bool has_revClass (Int edim) const;
-
-/** Takes input of model entity IDs, entity dimension, name of field and number
- * of components, to create a the rcField. This function
- * is used when fields are to be stored with mesh entities returned by
- * ask_revClass (Int edim, LOs class_ids)
- */
+  /** ask_revClass (Int edim, LOs class_ids): takes input of entity dimension
```
Comment (joshia5) [inline src/Omega_h_mesh.hpp:130]: is this fixing formatting?

File: src/Omega_h_mesh.hpp [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-#include <set>
-#include <string>
-#include <vector>
-
 #include <Omega_h_adj.hpp>
 #include <Omega_h_comm.hpp>
 #include <Omega_h_dist.hpp>
 #include <Omega_h_library.hpp>
 #include <Omega_h_tag.hpp>
+#include <array>
```
Comment (joshia5) [inline src/Omega_h_mesh.hpp:9]: is this trying to fix formatting?

**Comment by joshia5** [comment]
@jacobmerson in an hour or so on this, i couldn't spot any obvious issues. Though in my opinion micro commits may have helped to review changes which are formatting fixes, const correctness fixes, rc fixes, etc. But I defer to your judgement and since all tests pass, i am ok with merging this.

File: src/Omega_h_mesh.hpp (hunk 1 above)
Comment (jacobmerson) [reply src/Omega_h_mesh.hpp:130]: I'm guessing my editor inserted this based on the .clang-format but it definitely looks weird.

I can go through and remove all this added white space if you think I should.

**Comment by jacobmerson** [comment]
@joshia5  Thanks for taking a look. I agree I should have used more commits on the stuff I was working on. I was waiting to commit until I had something that passed the tests. And I ended up making a bunch of changes to make things work.

I think the many commits from upstream is correct.

I can try to undo the formatting changes that my editor made if you think I should...

Also, do you have permission to merge? If not, hopefully someone has ownership while Cameron is out.

File: src/Omega_h_mesh.cpp [hunk 3]
```diff
@@ -217,93 +220,87 @@ GO Mesh::nglobal_ents(Int ent_dim) {
...
   /* internal typically indicates migration/adaptation/file reading,
      when we do not want any invalidation to take place.
      the invalidation is there to prevent users changing coordinates
      etc. without updating dependent fields */
   if (!internal) react_to_set_tag(ent_dim, name);
-  tag->set_array(array);
-}
-
-template <typename T>
-void Mesh::add_tag(Topo_type ent_type, std::string const& name, Int ncomps,
```
Comment (joshia5) [inline src/Omega_h_mesh.cpp:270]: edited version not added

**Comment by jacobmerson** [comment]
#34 fixes the history issue we were having with this pull request.

**Pull request closed without merging**